sweetest/src/main/java/com/mysugr/sweetest/framework/context/DependenciesTestContext.kt

```diff
@@ -83,7 +99,46 @@ class DependenciesTestContext {
      *  [provide]. [provide] automatically requires that the automatically created instance is used.
      */
     fun provide(clazz: KClass<*>) {
-        val dependency = getDependencyConfiguration(clazz)
-        TestEnvironment.dependencies.states[dependency].mode = DependencyMode.REAL
+        val mode = DependencyMode.REAL
+        addDependency(clazz) { state, _ ->
+            state.mode = mode
+        }
+    }
+
+    /**
+     * Checks if the type [clazz] is already added via the global module configuration.
+     * If so, the found [DependencyConfiguration] is added to the [DependencyManager.states].
+     * If not, a [DependencyConfiguration] is created dynamically and also added.
+     */
+    private fun addDependency(clazz: KClass<*>, block: (DependencyState<out Any>, DependencyMode?) -> Unit) {
+        with(TestEnvironment.dependencies) {
+            assertDependency(clazz)
+            val dependencyState = getDependencyConfiguration(clazz)?.let { configuration ->
+                states[configuration]
+            } ?: run {
+                states[DependencyConfiguration(clazz, null, null, null)]
+            }
+
+            block(dependencyState, dependencyState.configuration.defaultDependencyMode)
+        }
+    }
+
+    /**
+     * Asserts if the type [clazz] is already added.
+     */
+    private fun assertDependency(clazz: KClass<*>) {
+        check(TestEnvironment.dependencies.states[clazz] == null) {
+            "Can't change dependency mode or ${clazz.simpleName}, it " +
+                "has already been set before and can't be changed afterwards"
+        }
+    }
+
+    /**
+     * Asserts if the configured mode matches the requested one.
+     */
+    private fun assertMode(clazz: KClass<*>, configurationMode: DependencyMode?, mode: DependencyMode) {
+        check(configurationMode == mode || configurationMode == null) {
+            "Class $clazz has been configured as ${configurationMode?.name} but is requested as ${mode.name}!"
```

Comment:
for the user it's most important to not only know what exactly has been gone wrong but also what he can do. can you give some hints in "sweetest style"? 😁 
also i think we should not talk about a `Class` (...has been configured) but rather a `Dependency`. it's stated like that mostly throughout the error messages.